arrow_util/src/string.rs

```diff
@@ -62,6 +63,26 @@ impl<K: AsPrimitive<usize> + FromPrimitive + Zero> PackedStringArray<K> {
         id
     }
 
+    /// Extends this [`PackedStringArray`] by `range` elements from `other`
+    pub fn extend_from_range(&mut self, other: &PackedStringArray<K>, range: Range<usize>) {
+        let first_offset = other.offsets[range.start].as_();
+        let end_offset = other.offsets[range.end].as_();
+
+        let insert_offset = self.storage.len();
+
+        self.storage
+            .push_str(&other.storage[first_offset..end_offset]);
+
+        self.offsets.extend(
+            (&other.offsets[(range.start + 1)..(range.end + 1)])
+                .iter()
+                .map(|x| {
+                    K::from_usize(x.as_() + insert_offset - first_offset)
```

Comment:
```suggestion
                    K::from_usize(x.as_() - first_offset + insert_offset)
```

This is less likely to overflow (on the other hand, having that many strings probably leads to a different issue :D )